src/service_discovery_setup.js

```diff
@@ -0,0 +1,274 @@
+      'settings. The error is: ${err}');
+  }
+};
+
+
+/**
+ * A small helper function.
+ *
+ * getmDNSconfig will look at settings that may be in the DB - set by
+ * the user and default settings. It will safely return a profile that
+ * will work for the calling application to use.
+ *
+ * @Returns: {Object} mDNS config options
+ *          e.g Options = { name: "Mozilla IoT Gateway",
+ *                         host: "mozillaGateway",     // Local DNS name
+ *                         txt: {  desc: "descriptive text",
+ *                                 protocols:  "http, https, oauth2 etc..",
+ *                                 power: "default power"
+ *                               }
+ *                       }
+ * This function searches /config or db.sqlite3 DB
+ *
+ * @param {}
+ */
+async function getmDNSconfig() {
+  let mDNSserviceDomain;
+
+  try {
+    mDNSserviceDomain = await Settings.get('localDNSname');
+
+    // There is no local service discovery name stored in the DB -
+    // use the default value
+    if (!mDNSserviceDomain) {
+      mDNSserviceDomain = await Settings.get('localDNSname');
+    }
+    const locName = config.get(
+      'settings.defaults.domain.localControl.mdnsServiceName');
+    const locProtocols = config.get(
+      'settings.defaults.domain.localControl.mdnsTxt.protocol');
+    const powerCons = config.get(
+      'settings.defaults.domain.localControl.mdnsTxt.power');
+    const description = config.get(
+      'settings.defaults.domain.localControl.mdnsTxt.desc');
+
+    const txt = {desc: description, protocols: locProtocols, power: powerCons};
+    const options = {name: locName, host: mDNSserviceDomain, txt};
+
+    // console.log("* Options are: " + options.name);
+
+    return options;
+  } catch (err) {
+    // We should never get here. Don't attempt to start service discovery
+    // and log an error message. But allow the application to carry on
+
+    console.error('The service discovery services could not find ' +
+      'configuration settings. The error is: ${err}');
+  }
+}
+
+
+/**
+ * A small helper function.
+ *
+ * @Returns: 'True' if mDNS service discovery has a default or user configured
+ * state requesting it to be started.
+ *
+ * @Returns: 'False' if default or user configured or no state if found.
+ *
+ *
+ * This function searches /config or db.sqlite3 DB
+ *
+ * @param {}
+ */
+async function getmDNSstate() {
+  let mDNSstate;
+  try {
+    mDNSstate = await Settings.get('multicastDNSstate');
+    return mDNSstate;
+  } catch (err) {
+    // Catch this DB error. Since we don't know what state the mDNS process
+    // should be in make sure it's off
+    console.error('Error getting DB entry for multicast DNS state from ' +
+      'the DB. Should be True or False', err);
+
+    return config.get('settings.defaults.domain.localAccess');
+  }
+}
+
+
+module.exports = {
+  server: server,
+  getDNSconfig: getmDNSconfig,
```

Comment:
nit: Why is this one not `getmDNSconfig`?

Reply:
i've changed.